src/dune_rules/melange/melange_rules.ml

```diff
@@ -241,6 +241,64 @@ let setup_emit_cmj_rules ~sctx ~dir ~scope ~expander ~dir_contents
   in
   Buildable_rules.with_lib_deps ctx compile_info ~dir ~f
 
+module Runtime_deps = struct
+  let to_action_builder ~expander dep_conf =
+    let runtime_deps =
+      let runtime_deps, _sandbox = Dep_conf_eval.unnamed ~expander dep_conf in
+      runtime_deps
+    in
+    Action_builder.memoize "melange runtime deps" runtime_deps
+
+  let eval ~expander (deps : Dep_conf.t list) =
+    let open Memo.O in
+    let builder = to_action_builder ~expander deps in
+    let+ paths, _ = Action_builder.run builder Lazy in
+    paths
+
+  let targets ~output deps =
+    List.map
+      ~f:(fun src ->
+        let dst =
+          match output with
+          | `Private_library_or_emit output_dir ->
+            let target =
+              src |> Path.as_in_build_dir_exn
+              |> Path.Build.drop_build_context_exn
+            in
+            Path.Build.append_source output_dir target
+          | `Public_library (lib_dir, output_dir) ->
+            lib_output_path ~output_dir ~lib_dir src
+        in
+        (src, dst))
+      deps
+end
+
+let setup_runtime_assets_rules sctx ~dir ~target_dir ~mode
+    ~(mel : Melange_stanzas.Emit.t) ~output =
+  let open Memo.O in
+  let* melange_files =
+    let* expander = Super_context.expander sctx ~dir in
+    Runtime_deps.eval ~expander mel.runtime_deps
+  in
+  let targets = Runtime_deps.targets ~output melange_files in
+  let+ () =
+    let loc = mel.loc in
+    Memo.parallel_iter targets ~f:(fun (src, dst) ->
+        Super_context.add_rule ~loc ~dir ~mode sctx
+          (Action_builder.copy ~src ~dst))
```

Comment:
What about symlinking?

Reply:
great idea, done in [bef37ff](https://github.com/ocaml/dune/pull/7199/commits/bef37ffb47d1133fb3864e7778726bca8d88fe41)